managed/services/backup/removal_service.go

```diff
@@ -31,44 +32,46 @@ import (
 type RemovalService struct {
 	l  *logrus.Entry
 	db *reform.DB
-	s3 s3
 }
 
 // NewRemovalService creates new backup removal service.
-func NewRemovalService(db *reform.DB, s3 s3) *RemovalService {
+func NewRemovalService(db *reform.DB) *RemovalService {
 	return &RemovalService{
 		l:  logrus.WithField("component", "services/backup/removal"),
 		db: db,
-		s3: s3,
 	}
 }
 
 // DeleteArtifact deletes specified artifact.
 func (s *RemovalService) DeleteArtifact(ctx context.Context, artifactID string, removeFiles bool) error {
-	artifactName, s3Config, err := s.beginDeletingArtifact(artifactID)
-	if err != nil {
-		return err
-	}
+	var artifact *models.Artifact
+	var err error
 
-	if s3Config != nil && removeFiles {
-		if err := s.s3.RemoveRecursive(
-			ctx,
-			s3Config.Endpoint,
-			s3Config.AccessKey,
-			s3Config.SecretKey,
-			s3Config.BucketName,
-			// Recursive listing finds all the objects with the specified prefix.
-			// There could be a problem e.g. when we have artifacts `backup-daily` and `backup-daily-1`, so
-			// listing by prefix `backup-daily` gives us both artifacts.
-			// To avoid such a situation we need to append a slash.
-			artifactName+"/"); err != nil {
-			if _, updateErr := models.UpdateArtifact(s.db.Querier, artifactID, models.UpdateArtifactParams{
-				Status: models.BackupStatusPointer(models.FailedToDeleteBackupStatus),
-			}); updateErr != nil {
-				s.l.WithError(updateErr).
-					Errorf("failed to set status %q for artifact %q", models.FailedToDeleteBackupStatus, artifactID)
-			}
+	if txErr := s.db.InTransactionContext(s.db.Querier.Context(), &sql.TxOptions{Isolation: sql.LevelSerializable}, func(tx *reform.TX) error {
```

Comment:
We name it like this everywhere:
txErr -> errTx

Reply:
Done.